core/web/solana_chains_controller.go

```diff
@@ -1,12 +1,37 @@
 package web
 
 import (
+	"context"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+
 	"github.com/smartcontractkit/chainlink/v2/core/services/chainlink"
 	"github.com/smartcontractkit/chainlink/v2/core/web/presenters"
 )
 
 func NewSolanaChainsController(app chainlink.Application) ChainsController {
-	chainSet := app.GetChains().Solana
+	chainSet := &relayerChainSet{app.GetChains().SolanaRelayer}
 	return newChainsController("solana", chainSet, ErrSolanaNotEnabled,
 		presenters.NewSolanaChainResource, app.GetLogger(), app.GetAuditLogger())
 }
+
+type relayerChainSet struct {
+	relayerFn func() (loop.Relayer, error)
+}
+
+func (r *relayerChainSet) ChainStatus(ctx context.Context, id string) (types.ChainStatus, error) {
+	relayer, err := r.relayerFn()
+	if err != nil {
+		return types.ChainStatus{}, err
+	}
+	return relayer.ChainStatus(ctx, id)
+}
+
+func (r *relayerChainSet) ChainStatuses(ctx context.Context, offset, limit int) ([]types.ChainStatus, int, error) {
+	relayer, err := r.relayerFn()
+	if err != nil {
+		return []types.ChainStatus{}, -1, err
```

Comment:
This is not meant to be meaningful. Do you prefer zero in these cases?
```suggestion
		return []types.ChainStatus{}, 0, err
```
I think the case for `-1` is that it's less likely to be misinterpreted up the stack by buggy code (which could see 0 as valid and move on e.g.), making the error more clear.

Reply:
-1 is a fine val, better than 0 for sure. my issue is that i can't tell what the int represents, so it would be easier to read like ~

const invalidXXX = -1

where XXX clues the reader into the intention of the return value

Reply:
Hmmm we usually default to the pattern that non-nil error means none of the other return args are valid, unless otherwise documented. Then we only have to document a few very exceptional cases.